### PR TITLE
Callback for basic TLS cert check

### PR DESCRIPTION
**Pull request by thtomate**
This callback at the moment only allows to check the sha512 hash. The comparison of the common name is not meaningful.
If the callback is not set there are still no checks! This should be changed in future.

I have to admit that the documentation is missing. Usage:

```
def print_cert(args):
    if args["sha512"] == "DF:A2:CE:07:EB:FA:F7:47:EA:56:7A:22:AD:6D:53:75:B7:BD:E3:…":
        return True
    else:
        print("Zert abgelehnt")
        return False

account.set_callbackCheckCert(print_cert)
```

### CODE REVIEW

**Comment by thtomate** [comment]
somehow it should be differed by gwenhywfar version
I don't know whether I have time to work on this the next days

**Comment by thtomate** [comment]
This could be fixed quite fast.
The usage should be reworked, as now the status of the check by gwenhywfar is accessable, which is mostly likely using the system trust store.

**Comment by julkah** [comment]
Why is it not merged yet?

**Comment by monofox** [comment]
Hey @julkah,

might be, that i didn't checked it yet due to lack of time.
Will have a look on weekend.

Regards,
Lukas

File: aqbanking/pyaqhandler.cpp [hunk 1]
```diff
@@ -98,11 +100,42 @@ int PyAqHandler::getPassword(uint32_t flags, const char *token, const char *titl
...
+		fprintf(stderr, "%s", "No answer on cert check returned!\n");
+		PyErr_Print();
+		return GWEN_ERROR_SSL_SECURITY;
+	} else {
+		int cert_valid = PyObject_IsTrue(result);
+		Py_DECREF(result);
+		return cert_valid? 0 : GWEN_ERROR_SSL_SECURITY;
+	}
+
 	return 0;
```
Comment (monofox) [inline aqbanking/pyaqhandler.cpp:139]:
```suggestion
```
return not required there anymore - as there is already ensured that a proper return is given (l110, l132, l136).

**Review by monofox — APPROVED** [review]

**Comment by monofox** [comment]
Thanks @thtomate for the contribution, nice work.

**Pull request merged**